e2e/pages/vm-sidebar.po.ts

```diff
@@ -171,4 +172,142 @@ export class VMSidebar extends CloudstackUiPage {
       by.cssContainingText('.value', expected),
     );
   }
+
+  clickChangeSO() {
+    const edit = element(by.css('cs-service-offering-details .mat-icon-button'));
+    const EC = protractor.ExpectedConditions;
+    browser.wait(EC.elementToBeClickable(edit), 2000, 'No edit Group button').then(() => {
+      edit.click();
+      this.waitDialogModal();
+    });
+  }
+
+  selectFixedTab() {
+    element
+      .all(by.css('cs-service-offering-dialog-container mat-button-toggle-group mat-button-toggle'))
+      .first()
+      .click();
+  }
+
+  changeSO() {
+    element(
+      by.xpath(
+        "//mat-radio-button[@class='mat-radio-button mat-accent mat-radio-checked']/ancestor::tbody//mat-radio-button[@class='mat-radio-button mat-accent']",
+      ),
+    ).click();
+  }
+
+  getSelectedSO() {
+    return element(
+      by.xpath(
+        "//mat-radio-button[@class='mat-radio-button mat-accent mat-radio-checked']/ancestor::td/preceding-sibling::td//span",
+      ),
+    ).getText();
+  }
+
+  getChangedSO() {
+    return element(
+      by.xpath(`//cs-service-offering-details//div[text()="Name"]/following-sibling::div`),
+    ).getText();
+  }
```

Comment:
As I understand this function returns the name of the current SO. In this case, such a name for the function is incorrect, since the function can return the name of not only a changed SO. Also, you do not receive the SO itself, but only its name.

Let's name the functions for their intended purpose. Here, for example, something like this: getSOName

If I'm wrong, correct me.